tests/i18n/test_extraction.py

```diff
@@ -1165,3 +1069,37 @@ def test_po_changed_with_new_strings(self):
             "This is a hitherto undiscovered translatable string.",
             po_contents,
         )
+
+class InvalidLocaleExtractionTests(ExtractorTests):
+
+    def setUp(self):
+        super().setUp()
+        # Ensure locale directory exists
+        Path("locale").mkdir(exist_ok=True)
+
+    def test_invalid_locale_script_code(self):
```

Comment:
Please move these tests next to the other invalid locale tests (e.g. next to `test_invalid_locale_lower_country`, giving `test_invalid_locale_mixed_case_region` or similar).